e2e/consultemplate/consultemplate_test.go

```diff
@@ -0,0 +1,547 @@
+	t.Cleanup(serviceLookupJobCleanup)
+	serviceLookupAllocID := serviceLookupJobSubmission.AllocID("nomad_provider_service_lookup")
+
+	// Ensure the listing (nomadServices) template function has found all
+	// services within the default namespace.
+	err = waitForTaskFile(serviceLookupAllocID, "test", "${NOMAD_TASK_DIR}/services.conf",
+		func(out string) bool {
+			if !strings.Contains(out, "service default-nomad-provider-service-primary [bar foo]") {
+				return false
+			}
+			if !strings.Contains(out, "service default-nomad-provider-service-secondary [baz buz]") {
+				return false
+			}
+			return !strings.Contains(out, "service platform-nomad-provider-service-secondary [baz buz]")
+		}, nil)
+	must.NoError(t, err)
+
+	// Ensure the direct service lookup has found the entry we expect.
+	expected := fmt.Sprintf("service default-nomad-provider-service-primary [bar foo] dc1 %s", serviceAllocID)
+	err = waitForTaskFile(serviceLookupAllocID, "test", "${NOMAD_TASK_DIR}/service.conf",
+		func(out string) bool {
+			return strings.Contains(out, expected)
+		},
+		nil)
+	must.NoError(t, err, must.Sprintf("expected: %v", expected))
+
+	// Scale the default namespaced service job in order to change the expected
+	// number of entries.
+	serviceJobSubmission.Rerun(jobs3.MutateJobSpec(func(spec string) string {
+		return strings.Replace(spec, "count = 1", "count = 3", 1)
+	}))
+
+	// Pull the allocation ID for the job, we use this to ensure this is found
+	// in the rendered template later on. Wrap this in a wait due to the
+	// eventual consistency around the service registration process.
+	serviceJobAllocs := []map[string]string{}
+	must.Wait(t, wait.InitialSuccess(
+		wait.BoolFunc(func() bool {
+			serviceJobAllocs, err = e2eutil.AllocsForJob(serviceJobSubmission.JobID(), "default")
+			must.NoError(t, err)
+			return len(serviceJobAllocs) == 3
+		}),
+		wait.Timeout(10*time.Second),
+		wait.Gap(200*time.Millisecond),
+	), must.Sprint("unexpected number of allocs found"))
+
+	// Track the expected entries, including the allocID to make this test
+	// actually valuable.
+	var expectedEntries []string
+	for _, allocs := range serviceJobAllocs {
+		e := fmt.Sprintf("service default-nomad-provider-service-primary [bar foo] dc1 %s", allocs["ID"])
+		expectedEntries = append(expectedEntries, e)
+	}
+
+	// Ensure the direct service lookup has the new entries we expect.
+	err = waitForTaskFile(serviceLookupAllocID, "test", "${NOMAD_TASK_DIR}/service.conf",
+		func(out string) bool {
+			for _, entry := range expectedEntries {
+				if !strings.Contains(out, entry) {
+					return false
+				}
+			}
+			return true
+		}, nil)
+	must.NoError(t, err, must.Sprintf("expected: %v", expectedEntries))
+}
+
+func waitForTaskFile(allocID, task, path string, test func(out string) bool, wc *e2eutil.WaitConfig) error {
+	var err error
+	var out string
+	interval, retries := wc.OrDefault()
+
+	testutil.WaitForResultRetries(retries, func() (bool, error) {
```

Comment:
not really important, but other things were switched to `must.Wait()` -- did this one just get missed?

Reply:
Yup. Fixed.